rust/ql/lib/codeql/rust/internal/TypeMention.qll

```diff
@@ -106,6 +131,13 @@ class TypeParamMention extends TypeMention, TypeParam {
   override Type resolveType() { result = TTypeParamTypeParameter(this) }
 }
 
+// Used to represent implicit associated type type arguments in traits.
+class TypeAliasMention extends TypeMention, TypeAlias {
+  override TypeReprMention getTypeArgument(int i) { none() }
+
+  override Type resolveType() { result = TAssociatedTypeTypeParameter(this) }
```

Comment:
Not all `TypeAlias`es should be type mentions, so perhaps
```ql
  private Type t;

  TypeAliasMention() { t = TAssociatedTypeTypeParameter(this) }

  override TypeReprMention getTypeArgument(int i) { none() }

  override Type resolveType() { result = t }
```

Reply:
Yes, good point!